lib/tests/test_revset.rs

```diff
@@ -410,6 +410,176 @@ fn test_resolve_symbol_checkout(use_git: bool) {
     );
 }
 
+#[test]
+fn test_resolve_symbol_branches() {
+    let settings = testutils::user_settings();
+    let test_repo = TestRepo::init(true);
+    let repo = &test_repo.repo;
+
+    let mut tx = repo.start_transaction(&settings, "test");
+    let mut_repo = tx.mut_repo();
+
+    let commit1 = write_random_commit(mut_repo, &settings);
+    let commit2 = write_random_commit(mut_repo, &settings);
+    let commit3 = write_random_commit(mut_repo, &settings);
+    let commit4 = write_random_commit(mut_repo, &settings);
+    let commit5 = write_random_commit(mut_repo, &settings);
+
+    mut_repo.set_local_branch("local".to_owned(), RefTarget::Normal(commit1.id().clone()));
+    mut_repo.set_remote_branch(
+        "remote".to_owned(),
+        "origin".to_owned(),
+        RefTarget::Normal(commit2.id().clone()),
+    );
+    mut_repo.set_local_branch(
+        "local-remote".to_owned(),
+        RefTarget::Normal(commit3.id().clone()),
+    );
+    mut_repo.set_remote_branch(
+        "local-remote".to_owned(),
+        "origin".to_owned(),
+        RefTarget::Normal(commit4.id().clone()),
+    );
+    mut_repo.set_remote_branch(
+        "local-remote".to_owned(),
+        "mirror".to_owned(),
+        mut_repo.get_local_branch("local-remote").unwrap(),
+    );
+
+    mut_repo.set_local_branch(
+        "local-conflicted".to_owned(),
+        RefTarget::Conflict {
+            removes: vec![commit1.id().clone()],
+            adds: vec![commit3.id().clone(), commit2.id().clone()],
+        },
+    );
+    mut_repo.set_remote_branch(
+        "remote-conflicted".to_owned(),
+        "origin".to_owned(),
+        RefTarget::Conflict {
+            removes: vec![commit3.id().clone()],
+            adds: vec![commit5.id().clone(), commit4.id().clone()],
+        },
+    );
+
+    // Local only
+    assert_eq!(
+        resolve_symbol(mut_repo, "local", None).unwrap(),
+        vec![commit1.id().clone()],
+    );
+    insta::assert_debug_snapshot!(
+        resolve_symbol(mut_repo, "local@origin", None).unwrap_err(), @r###"
+    NoSuchRevision {
+        name: "local@origin",
+        candidates: [
+            "local",
+        ],
+    }
+    "###);
+
+    // Remote only
```

Comment:
Optional: I'd rename this branch to `locally-deleted`, or at least change the comment to something like "Remote only, like a branch that was fetched and then deleted"

Reply:
Added `(or locally deleted)` to the comment. Renaming branch can affect the suggestion (because string distance changes), and I don't wanna be bothered to update snapshots. :)